Panels/Panels.js

```diff
@@ -38,25 +157,78 @@ const PanelsBase = kind({
 			if (orientation === 'vertical') return VerticalArranger;
 			else return HorizontalArranger;
 		},
-		enteringProp: ({noAnimation}) => noAnimation ? null : 'hideChildren'
+		childProps: ({childProps, controls, id, noCloseButton}) => {
+			if ((noCloseButton && !controls) || !id) {
+				return childProps;
+			}
+
+			const updatedChildProps = Object.assign({}, childProps);
+			const controlsId = getControlsId(id);
+			const owns = updatedChildProps['aria-owns'];
+
+			if (owns) {
+				updatedChildProps['aria-owns'] = `${owns} ${controlsId}`;
+			} else {
+				updatedChildProps['aria-owns'] = controlsId;
+			}
+
+			return updatedChildProps;
+		},
+		className: ({controls, noCloseButton, styler}) => styler.append({
+			'agate-panels-hasControls': (!noCloseButton || !!controls) // If there is a close button or controls were specified
+		}),
+		style: ({controlsMeasurements, style = {}}) => (controlsMeasurements ? {
+			...style,
+			'--agate-panels-controls-width': controlsMeasurements.width + 'px'
+		} : style),
+		viewportId: ({id}) => id && `${id}-viewport`
 	},
-	render: ({children, ...props}) => {
-		const mappedChildren = mapChildren(children);
+	render: ({arranger, childProps, children, closeButtonAriaLabel, controls, controlsRef, generateId, id, index, noAnimation, noCloseButton, noSharedState, onApplicationClose, viewportId, ...rest}) => {
+		delete rest.controlsMeasurements;
+		delete rest.onBack;
+
+		const controlsId = getControlsId(id);
 
 		return (
-			<ViewManager
-				{...props}
-			>
-				{mappedChildren}
-			</ViewManager>
+			<div {...rest} id={id}>
+				<Controls
+					closeButtonAriaLabel={closeButtonAriaLabel}
+					id={controlsId}
+					spotlightId={controlsId}
+					noCloseButton={noCloseButton}
+					onApplicationClose={onApplicationClose}
+					ref={controlsRef}
+				>
+					{controls}
+				</Controls>
+				<Viewport
+					arranger={arranger}
+					childProps={childProps}
+					generateId={generateId}
+					id={viewportId}
+					index={index}
+					noAnimation={noAnimation}
+					noSharedState={noSharedState}
+				>
+					{children}
+				</Viewport>
+			</div>
 		);
 	}
 });
 
+const PanelsDecorator = compose(
+	Slottable({slots: ['controls']}),
+	CancelDecorator({cancel: 'onBack'}),
+	Measurable({refProp: 'controlsRef', measurementProp: 'controlsMeasurements'}),
+	IdProvider
+);
+
+const Panels = PanelsDecorator(PanelsBase);
+
 export default PanelsBase;
```

Comment:
Should the default export be `Panels` now and only let devs get `PanelsBase` by name?

Reply:
Good observation 👍